models/users.js

```diff
@@ -572,6 +572,32 @@ const fetchAllUsers = async () => {
   return users;
 };
 
+/**
+ *
+ * @param {[string]} userIds  Array id's of user
+ * @returns Object containing the details of the users whose userId was provided.
+ */
+const fetchUserByIds = async (userIds = []) => {
+  if (userIds.length === 0) {
+    return {};
+  }
+  try {
+    const users = {};
+    const usersRefs = userIds.map((docId) => userModel.doc(docId));
+    const documents = await firestore.getAll(...usersRefs);
```

Comment:
here in the line of code where you are getting the user data from Firestore. The code is:

const documents = await firestore.getAll(...usersRefs);
The problem with this code is that it is using the getAll() method to get all of the documents in the usersRefs array. However, the getAll() method can be very expensive, especially if there are a lot of documents in the array.

To fix this bug, you can use the get() method instead of the getAll() method. The get() method will only get the document that you specify, so it will be much faster.

Reply:
Regarding the getAll method, it fetches only the details specified in the parameters, as documented [here](https://googleapis.dev/nodejs/firestore/latest/Firestore.html#getAll).

Additionally, it's important to note that there is no get method available directly on the Firestore object.

If you were referring to using the get method of models, that won't work in this case because it requires conditions using where, and the where clause works with document keys (e.g., discordId in [...]).

Since the requirement is to fetch user data with document IDs, using getAll seems to be the appropriate approach in this scenario.

Note: this is my first time working with Firestore, so I might have limited knowledge. If you have any other suggestions or more efficient ways to handle this, I'm open to learning and making improvements.

Thank you for your feedback and patience!